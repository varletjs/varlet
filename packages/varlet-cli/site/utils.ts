import { get } from 'lodash-es'
import { StyleProvider, type Theme, withSiteConfigNamespace } from '@varlet/cli/client'

export interface Menu {
  doc: string
  text: Record<string, string>
  type: MenuTypes
  useMobile?: boolean
}

export enum MenuTypes {
  TITLE = 1,
  COMPONENT = 2,
  DOCUMENTATION = 3,
}

export function removeEmpty(object: Record<string, string> = {}) {
  return Object.keys(object).reduce((record: Record<string, string>, key) => {
    const value = object[key]
    value && (record[key] = value)
    return record
  }, {})
}

export function isPhone() {
  return /Android|webOS|iPhone|iPod|BlackBerry|Pad/i.test(navigator.userAgent)
}

export function inIframe() {
  return window.self !== window.top
}

export function setTheme(config: Record<string, any>, name: Theme) {
  const styleVars = withSiteConfigNamespace(get(config, name, {}))

  StyleProvider(styleVars)
}

export function utoa(data: string): string {
  return btoa(unescape(encodeURIComponent(data)))
}
