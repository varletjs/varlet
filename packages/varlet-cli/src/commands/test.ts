import execa from 'execa'
import { VITEST_CONFIG } from '../shared/constant'
import { buildSiteEntry } from '../compiler/compileSiteEntry'

interface TestCommandOptions {
  watch?: boolean
  component?: string
  coverage?: boolean
}

export async function test({ component, watch, coverage }: TestCommandOptions) {
  process.env.NODE_ENV = 'test'

  // Test cases may contain documentation example code, and site dependencies need to build
  await buildSiteEntry()

  const args = ['--config', VITEST_CONFIG]

  if (!watch) {
    args.unshift('run')
  }

  if (coverage) {
    args.push('--coverage')
  }

  if (component) {
    args.push('--dir', `src/${component.trim()}`)
  }

  await execa('vitest', args, { stdin: 'inherit', stdout: 'inherit', stderr: 'inherit' })
}
