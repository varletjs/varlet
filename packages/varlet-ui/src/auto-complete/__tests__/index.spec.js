import { createApp, h, nextTick } from 'vue'
import { mount } from '@vue/test-utils'
import { expect, test, vi } from 'vitest'
import { z } from 'zod'
import AutoComplete from '..'
import { trigger, triggerKeyboard } from '../../utils/test'

test('auto-complete plugin', () => {
  const app = createApp({}).use(AutoComplete)
  expect(app.component(AutoComplete.name)).toBeTruthy()
})

test('auto-complete modelValue', async () => {
  const onUpdateModelValue = vi.fn()
  const wrapper = mount(AutoComplete, {
    props: {
      modelValue: '',
      'onUpdate:modelValue': onUpdateModelValue,
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  await wrapper.find('input').setValue('a')
  expect(onUpdateModelValue).toHaveBeenCalledWith('a')
  wrapper.unmount()
})

test('auto-complete options', async () => {
  const onUpdateModelValue = vi.fn()
  const wrapper = mount(AutoComplete, {
    props: {
      modelValue: 'a',
      'onUpdate:modelValue': onUpdateModelValue,
      options: [
        {
          value: 'Option 1',
          label: 'Option 1',
        },
        {
          value: 'Option 2',
          label: 'Option 2',
        },
      ],
    },
  })

  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')
  await wrapper.trigger('focusin')
  expect(document.querySelector('.var-menu__menu').style.display).toBe('')

  await trigger(document.querySelector('.var-menu-option'), 'click')
  expect(onUpdateModelValue).toHaveBeenCalledWith('Option 1')
  wrapper.unmount()
})

test('auto-complete getShow', async () => {
  const wrapper = mount(AutoComplete, {
    props: {
      modelValue: 'a',
      options: [
        {
          value: 'Option 1',
          label: 'Option 1',
        },
        {
          value: 'Option 2',
          label: 'Option 2',
        },
      ],
      getShow: (value) => value === 'b',
    },
  })

  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')
  await wrapper.trigger('focusin')
  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')
  await wrapper.find('input').trigger('blur')

  await wrapper.setProps({ modelValue: 'b' })
  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')
  await wrapper.trigger('focusin')
  expect(document.querySelector('.var-menu__menu').style.display).toBe('')

  wrapper.unmount()
})

test('auto-complete options custom key', async () => {
  const onUpdateModelValue = vi.fn()
  const wrapper = mount(AutoComplete, {
    props: {
      modelValue: 'a',
      labelKey: 'name',
      valueKey: 'id',
      'onUpdate:modelValue': onUpdateModelValue,
      options: [
        {
          id: 'Option Id 1',
          name: 'Option Name 1',
        },
        {
          id: 'Option Id 2',
          name: 'Option Name 2',
        },
      ],
    },
  })

  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')
  await wrapper.trigger('focusin')
  expect(document.querySelector('.var-menu__menu').style.display).toBe('')

  await trigger(document.querySelector('.var-menu-option'), 'click')
  expect(onUpdateModelValue).toHaveBeenCalledWith('Option Id 1')
  wrapper.unmount()
})

test('auto-complete placeholder', () => {
  const wrapper = mount(AutoComplete, {
    props: {
      placeholder: 'Hello',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('auto-complete maxlength', async () => {
  const onUpdateModelValue = vi.fn()
  const wrapper = mount(AutoComplete, {
    props: {
      maxlength: 1,
      'onUpdate:modelValue': onUpdateModelValue,
    },
  })

  await wrapper.find('input').setValue('123')
  expect(onUpdateModelValue).toHaveBeenCalledWith('1')

  wrapper.unmount()
})

test('auto-complete maxlength for handle auto complete', async () => {
  const onUpdateModelValue = vi.fn()
  const wrapper = mount(AutoComplete, {
    props: {
      modelValue: 'a',
      maxlength: 3,
      options: [
        {
          value: 'Option 1',
          label: 'Option 1',
        },
        {
          value: 'Option 2',
          label: 'Option 2',
        },
      ],
      'onUpdate:modelValue': onUpdateModelValue,
    },
  })

  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')
  await wrapper.trigger('focusin')
  expect(document.querySelector('.var-menu__menu').style.display).toBe('')

  await trigger(document.querySelector('.var-menu-option'), 'click')
  expect(onUpdateModelValue).toHaveBeenCalledWith('Opt')

  wrapper.unmount()
})

test('auto-complete disabled', async () => {
  const onUpdateModelValue = vi.fn()
  const onClick = vi.fn()
  const wrapper = mount(AutoComplete, {
    props: {
      disabled: true,
      modelValue: 'a',
      options: [
        {
          value: 'Option 1',
          label: 'Option 1',
        },
        {
          value: 'Option 2',
          label: 'Option 2',
        },
      ],
      onClick,
      'onUpdate:modelValue': onUpdateModelValue,
    },
  })

  expect(wrapper.html()).toMatchSnapshot()

  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')
  await wrapper.trigger('focusin')
  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')

  await wrapper.trigger('click')
  expect(onClick).not.toHaveBeenCalled()

  wrapper.unmount()
})

test('auto-complete readonly', async () => {
  const onUpdateModelValue = vi.fn()
  const onClick = vi.fn()
  const wrapper = mount(AutoComplete, {
    props: {
      readonly: true,
      modelValue: 'a',
      options: [
        {
          value: 'Option 1',
          label: 'Option 1',
        },
        {
          value: 'Option 2',
          label: 'Option 2',
        },
      ],
      onClick,
      'onUpdate:modelValue': onUpdateModelValue,
    },
  })

  expect(wrapper.html()).toMatchSnapshot()

  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')
  await wrapper.trigger('focusin')
  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')

  await wrapper.trigger('click')
  expect(onClick).toBeCalledTimes(1)

  wrapper.unmount()
})

test('auto-complete clearable', async () => {
  const onClear = vi.fn()
  const onUpdateModelValue = vi.fn()
  const wrapper = mount(AutoComplete, {
    props: {
      modelValue: 'a',
      clearable: true,
      onClear,
      'onUpdate:modelValue': onUpdateModelValue,
    },
  })

  await wrapper.find('.var-icon-close-circle').trigger('click')
  expect(onClear).toHaveBeenCalledWith('')
  expect(onUpdateModelValue).toHaveBeenCalledWith('')
  wrapper.unmount()
})

test('auto complete onInput onChange onFocus onBlur', async () => {
  const onUpdateModelValue = vi.fn()
  const onInput = vi.fn()
  const onChange = vi.fn()
  const onFocus = vi.fn()
  const onBlur = vi.fn()

  const wrapper = mount(AutoComplete, {
    props: {
      modelValue: '',
      'onUpdate:modelValue': onUpdateModelValue,
      onInput,
      onChange,
      onFocus,
      onBlur,
    },
  })

  await wrapper.trigger('focusin')
  expect(onFocus).toHaveBeenCalledTimes(1)
  await wrapper.find('input').trigger('blur')
  expect(onBlur).toHaveBeenCalledTimes(1)

  await wrapper.find('input').setValue('a')
  expect(onInput).toHaveBeenCalledTimes(1)
  expect(onChange).toHaveBeenCalledTimes(1)

  wrapper.unmount()
})

test('auto-complete hint and line', () => {
  const wrapper = mount(AutoComplete, {
    props: {
      hint: false,
      line: false,
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('auto-complete custom color', () => {
  const wrapper = mount(AutoComplete, {
    props: {
      textColor: 'red',
      focusColor: 'blue',
      blurColor: 'green',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('auto-complete size', () => {
  const wrapper = mount(AutoComplete, {
    props: {
      size: 'small',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('auto-complete variant', () => {
  const wrapper = mount(AutoComplete, {
    props: {
      variant: 'outlined',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('auto-complete slots', () => {
  const wrapper = mount(AutoComplete, {
    slots: {
      'append-icon': () => 'append-icon',
      'prepend-icon': () => 'prepend-icon',
      'clear-icon': () => 'clear-icon',
      'extra-message': () => 'extra-message',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('auto-complete validation', async () => {
  const onUpdateModelValue = vi.fn()
  const wrapper = mount(AutoComplete, {
    props: {
      modelValue: '',
      rules: [(v) => !!v || 'Required'],
      'onUpdate:modelValue': onUpdateModelValue,
    },
  })

  await wrapper.vm.validate()
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.vm.resetValidation()
  await nextTick()
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.vm.reset()
  await nextTick()
  expect(onUpdateModelValue).toHaveBeenCalledTimes(1)
  wrapper.unmount()
})

test('auto-complete click outside', async () => {
  const wrapper = mount(AutoComplete, {
    props: {
      modelValue: 'a',
      options: [
        {
          value: 'Option 1',
          label: 'Option 1',
        },
        {
          value: 'Option 2',
          label: 'Option 2',
        },
      ],
    },
  })

  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')
  await wrapper.trigger('focusin')
  expect(document.querySelector('.var-menu__menu').style.display).toBe('')

  await trigger(document, 'click')
  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')

  wrapper.unmount()
})

test('auto-complete select option by keyboard', async () => {
  const onUpdateModelValue = vi.fn()
  const wrapper = mount(AutoComplete, {
    props: {
      modelValue: 'a',
      'onUpdate:modelValue': onUpdateModelValue,
      options: [
        {
          value: 'Option 1',
          label: 'Option 1',
        },
        {
          value: 'Option 2',
          label: 'Option 2',
        },
      ],
    },
  })

  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')
  await wrapper.trigger('focusin')
  expect(document.querySelector('.var-menu__menu').style.display).toBe('')

  await triggerKeyboard(window, 'keydown', { key: 'ArrowDown' })
  await trigger(document.querySelector('.var-menu-option'), 'focus')
  await triggerKeyboard(window, 'keydown', { key: 'Enter' })
  expect(onUpdateModelValue).toHaveBeenCalledWith('Option 1')

  wrapper.unmount()
})

test('auto-complete close menu by keyboard tab', async () => {
  const wrapper = mount(AutoComplete, {
    props: {
      modelValue: 'a',
      options: [
        {
          value: 'Option 1',
          label: 'Option 1',
        },
        {
          value: 'Option 2',
          label: 'Option 2',
        },
      ],
    },
  })

  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')
  await wrapper.trigger('focusin')
  expect(document.querySelector('.var-menu__menu').style.display).toBe('')

  await triggerKeyboard(window, 'keydown', { key: 'Tab' })
  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')

  wrapper.unmount()
})

test('auto-complete handle same value by option', async () => {
  const onChange = vi.fn()
  const wrapper = mount(AutoComplete, {
    props: {
      modelValue: 'a',
      options: [
        {
          value: 'a',
          label: 'a',
        },
      ],
      onChange,
    },
  })

  expect(document.querySelector('.var-menu__menu').style.display).toBe('none')
  await wrapper.trigger('focusin')
  expect(document.querySelector('.var-menu__menu').style.display).toBe('')

  await trigger(document.querySelector('.var-menu-option'), 'click')
  expect(onChange).toHaveBeenCalledTimes(0)

  wrapper.unmount()
})

test('auto-complete focus and blur methods', () => {
  const onFocus = vi.fn()
  const onBlur = vi.fn()
  const wrapper = mount(AutoComplete, {
    props: {
      onFocus,
      onBlur,
    },
  })

  wrapper.vm.focus()
  expect(onFocus).toHaveBeenCalledTimes(1)

  wrapper.vm.blur()
  expect(onBlur).toHaveBeenCalledTimes(1)

  wrapper.unmount()
})

test('auto-complete clear-icon scoped clear method', async () => {
  const onClear = vi.fn()
  const wrapper = mount(AutoComplete, {
    props: {
      modelValue: 'a',
      clearable: true,
      onClear,
    },
    slots: {
      'clear-icon': ({ clear }) =>
        h('div', [h('div', { class: 'target', onClick: clear }), h('div', { class: 'other' })]),
    },
  })

  await wrapper.find('.other').trigger('click')
  expect(onClear).toHaveBeenCalledTimes(0)

  await wrapper.find('.target').trigger('click')
  expect(onClear).toHaveBeenCalledTimes(1)

  wrapper.unmount()
})

test('auto-complete blur without focus', () => {
  const onBlur = vi.fn()
  const wrapper = mount(AutoComplete, {
    props: {
      onBlur,
    },
  })

  wrapper.vm.blur()
  expect(onBlur).toHaveBeenCalledTimes(0)

  wrapper.vm.focus()
  wrapper.vm.blur()
  expect(onBlur).toHaveBeenCalledTimes(1)
  wrapper.unmount()
})

test('auto-complete valiation with zod', async () => {
  const onUpdateModelValue = vi.fn()
  const wrapper = mount(AutoComplete, {
    props: {
      modelValue: '',
      rules: z.string().min(1, 'Required'),
      'onUpdate:modelValue': onUpdateModelValue,
    },
  })

  await wrapper.vm.validate()
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.vm.resetValidation()
  await nextTick()
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.vm.reset()
  await nextTick()
  expect(onUpdateModelValue).toHaveBeenCalledTimes(1)
  wrapper.unmount()
})
