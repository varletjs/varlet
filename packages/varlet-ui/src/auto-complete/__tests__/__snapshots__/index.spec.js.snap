// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`auto-complete custom color 1`] = `
"<div class="var-auto-complete" tabindex="0">
  <div class="var-menu var--box var-menu-select var-auto-complete__menu-select" tabindex="-1" auto-complete-cover="">
    <div class="var-input var--box">
      <div class="var-field-decorator var--box var-field-decorator--standard">
        <div class="var-field-decorator__controller" style="color: green; cursor: text; overflow: hidden; --field-decorator-middle-offset-left: 0px; --field-decorator-middle-offset-width: 0px; --field-decorator-middle-offset-height: 0px;">
          <div class="var-field-decorator__icon"></div>
          <div class="var-field-decorator__middle">
            <!--v-if--><input id="var-input-mock-id" class="var-input__input" autocomplete="off" type="text" style="color: red; caret-color: blue;">
          </div>
          <!--v-if-->
          <div class="var-field-decorator__icon">
            <!--v-if-->
          </div>
        </div>
        <div class="var-field-decorator__line" style="background: green;">
          <div class="var-field-decorator__dot" style="background: blue;"></div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <transition-stub name="var-form-details" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`auto-complete disabled 1`] = `
"<div class="var-auto-complete">
  <div class="var-menu var--box var-menu-select var-auto-complete__menu-select" tabindex="-1" auto-complete-cover="">
    <div class="var-input var--box">
      <div class="var-field-decorator var--box var-field-decorator--standard var-field-decorator--disabled">
        <div class="var-field-decorator__controller var-field-decorator--disabled" style="overflow: visible; --field-decorator-middle-offset-left: 0px; --field-decorator-middle-offset-width: 0px; --field-decorator-middle-offset-height: 0px;">
          <div class="var-field-decorator__icon"></div>
          <div class="var-field-decorator__middle">
            <!--v-if--><input id="var-input-mock-id" class="var-input__input var-input--disabled" autocomplete="off" disabled="" type="text">
          </div>
          <!--v-if-->
          <div class="var-field-decorator__icon">
            <!--v-if-->
          </div>
        </div>
        <div class="var-field-decorator__line var-field-decorator--line-disabled">
          <div class="var-field-decorator__dot var-field-decorator--line-disabled"></div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <transition-stub name="var-form-details" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`auto-complete hint and line 1`] = `
"<div class="var-auto-complete" tabindex="0">
  <div class="var-menu var--box var-menu-select var-auto-complete__menu-select" tabindex="-1" auto-complete-cover="">
    <div class="var-input var--box">
      <div class="var-field-decorator var--box var-field-decorator--standard">
        <div class="var-field-decorator__controller" style="cursor: text; overflow: hidden; --field-decorator-middle-offset-left: 0px; --field-decorator-middle-offset-width: 0px; --field-decorator-middle-offset-height: 0px;">
          <div class="var-field-decorator__icon var-field-decorator--icon-non-hint"></div>
          <div class="var-field-decorator__middle var-field-decorator--middle-non-hint">
            <!--v-if--><input id="var-input-mock-id" class="var-input__input" autocomplete="off" type="text" style="--input-placeholder-color: var(--field-decorator-placeholder-color, var(--field-decorator-blur-color));">
          </div>
          <!--v-if-->
          <div class="var-field-decorator__icon var-field-decorator--icon-non-hint">
            <!--v-if-->
          </div>
        </div>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <transition-stub name="var-form-details" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`auto-complete modelValue 1`] = `
"<div class="var-auto-complete" tabindex="0">
  <div class="var-menu var--box var-menu-select var-auto-complete__menu-select" tabindex="-1" auto-complete-cover="">
    <div class="var-input var--box">
      <div class="var-field-decorator var--box var-field-decorator--standard">
        <div class="var-field-decorator__controller" style="cursor: text; overflow: hidden; --field-decorator-middle-offset-left: 0px; --field-decorator-middle-offset-width: 0px; --field-decorator-middle-offset-height: 0px;">
          <div class="var-field-decorator__icon"></div>
          <div class="var-field-decorator__middle">
            <!--v-if--><input id="var-input-mock-id" class="var-input__input" autocomplete="off" type="text">
          </div>
          <!--v-if-->
          <div class="var-field-decorator__icon">
            <!--v-if-->
          </div>
        </div>
        <div class="var-field-decorator__line">
          <div class="var-field-decorator__dot"></div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <transition-stub name="var-form-details" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`auto-complete placeholder 1`] = `
"<div class="var-auto-complete" tabindex="0">
  <div class="var-menu var--box var-menu-select var-auto-complete__menu-select" tabindex="-1" auto-complete-cover="">
    <div class="var-input var--box">
      <div class="var-field-decorator var--box var-field-decorator--standard">
        <div class="var-field-decorator__controller" style="cursor: text; overflow: hidden; --field-decorator-middle-offset-left: 0px; --field-decorator-middle-offset-width: 0px; --field-decorator-middle-offset-height: 0px;">
          <div class="var-field-decorator__icon"></div>
          <div class="var-field-decorator__middle">
            <!--v-if--><input id="var-input-mock-id" class="var-input__input" autocomplete="off" type="text">
          </div><label class="var-field-decorator__placeholder var--ellipsis var-field-decorator--hint-center var-field-decorator--transition-disabled" for="var-input-mock-id"><span>Hello</span></label>
          <div class="var-field-decorator__icon">
            <!--v-if-->
          </div>
        </div>
        <div class="var-field-decorator__line">
          <div class="var-field-decorator__dot"></div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <transition-stub name="var-form-details" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`auto-complete readonly 1`] = `
"<div class="var-auto-complete" tabindex="0">
  <div class="var-menu var--box var-menu-select var-auto-complete__menu-select" tabindex="-1" auto-complete-cover="">
    <div class="var-input var--box">
      <div class="var-field-decorator var--box var-field-decorator--standard">
        <div class="var-field-decorator__controller" style="overflow: visible; --field-decorator-middle-offset-left: 0px; --field-decorator-middle-offset-width: 0px; --field-decorator-middle-offset-height: 0px;">
          <div class="var-field-decorator__icon"></div>
          <div class="var-field-decorator__middle">
            <!--v-if--><input id="var-input-mock-id" class="var-input__input" autocomplete="off" readonly="" type="text">
          </div>
          <!--v-if-->
          <div class="var-field-decorator__icon">
            <!--v-if-->
          </div>
        </div>
        <div class="var-field-decorator__line">
          <div class="var-field-decorator__dot"></div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <transition-stub name="var-form-details" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`auto-complete size 1`] = `
"<div class="var-auto-complete" tabindex="0">
  <div class="var-menu var--box var-menu-select var-auto-complete__menu-select" tabindex="-1" auto-complete-cover="">
    <div class="var-input var--box">
      <div class="var-field-decorator var--box var-field-decorator--standard var-field-decorator--small">
        <div class="var-field-decorator__controller" style="cursor: text; overflow: hidden; --field-decorator-middle-offset-left: 0px; --field-decorator-middle-offset-width: 0px; --field-decorator-middle-offset-height: 0px;">
          <div class="var-field-decorator__icon"></div>
          <div class="var-field-decorator__middle">
            <!--v-if--><input id="var-input-mock-id" class="var-input__input" autocomplete="off" type="text">
          </div>
          <!--v-if-->
          <div class="var-field-decorator__icon">
            <!--v-if-->
          </div>
        </div>
        <div class="var-field-decorator__line">
          <div class="var-field-decorator__dot"></div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <transition-stub name="var-form-details" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`auto-complete slots 1`] = `
"<div class="var-auto-complete" tabindex="0">
  <div class="var-menu var--box var-menu-select var-auto-complete__menu-select" tabindex="-1" auto-complete-cover="">
    <div class="var-input var--box">
      <div class="var-field-decorator var--box var-field-decorator--standard">
        <div class="var-field-decorator__controller" style="cursor: text; overflow: hidden; --field-decorator-middle-offset-left: 0px; --field-decorator-middle-offset-width: 0px; --field-decorator-middle-offset-height: 0px;">
          <div class="var-field-decorator__icon">prepend-icon</div>
          <div class="var-field-decorator__middle">
            <!--v-if--><input id="var-input-mock-id" class="var-input__input" autocomplete="off" type="text">
          </div>
          <!--v-if-->
          <div class="var-field-decorator__icon">
            <!--v-if-->append-icon
          </div>
        </div>
        <div class="var-field-decorator__line">
          <div class="var-field-decorator__dot"></div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <transition-stub name="var-form-details" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`auto-complete valiation with zod 1`] = `
"<div class="var-auto-complete" tabindex="0">
  <div class="var-menu var--box var-menu-select var-auto-complete__menu-select" tabindex="-1" auto-complete-cover="">
    <div class="var-input var--box">
      <div class="var-field-decorator var--box var-field-decorator--standard">
        <div class="var-field-decorator__controller var-field-decorator--error" style="cursor: text; overflow: hidden; --field-decorator-middle-offset-left: 0px; --field-decorator-middle-offset-width: 0px; --field-decorator-middle-offset-height: 0px;">
          <div class="var-field-decorator__icon"></div>
          <div class="var-field-decorator__middle">
            <!--v-if--><input id="var-input-mock-id" class="var-input__input" autocomplete="off" type="text">
          </div>
          <!--v-if-->
          <div class="var-field-decorator__icon">
            <!--v-if-->
          </div>
        </div>
        <div class="var-field-decorator__line var-field-decorator--line-error">
          <div class="var-field-decorator__dot var-field-decorator--line-error"></div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <transition-stub name="var-form-details" appear="false" persisted="false" css="true">
    <div class="var-form-details">
      <div class="var-form-details__error-message">
        <transition-stub name="var-form-details__message" appear="false" persisted="false" css="true">
          <div>Required</div>
        </transition-stub>
      </div>
      <div class="var-form-details__extra-message">
        <transition-stub name="var-form-details__message" appear="false" persisted="false" css="true">
          <!--v-if-->
        </transition-stub>
      </div>
    </div>
  </transition-stub>
</div>"
`;

exports[`auto-complete valiation with zod 2`] = `
"<div class="var-auto-complete" tabindex="0">
  <div class="var-menu var--box var-menu-select var-auto-complete__menu-select" tabindex="-1" auto-complete-cover="">
    <div class="var-input var--box">
      <div class="var-field-decorator var--box var-field-decorator--standard">
        <div class="var-field-decorator__controller" style="cursor: text; overflow: hidden; --field-decorator-middle-offset-left: 0px; --field-decorator-middle-offset-width: 0px; --field-decorator-middle-offset-height: 0px;">
          <div class="var-field-decorator__icon"></div>
          <div class="var-field-decorator__middle">
            <!--v-if--><input id="var-input-mock-id" class="var-input__input" autocomplete="off" type="text">
          </div>
          <!--v-if-->
          <div class="var-field-decorator__icon">
            <!--v-if-->
          </div>
        </div>
        <div class="var-field-decorator__line">
          <div class="var-field-decorator__dot"></div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <transition-stub name="var-form-details" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`auto-complete validation 1`] = `
"<div class="var-auto-complete" tabindex="0">
  <div class="var-menu var--box var-menu-select var-auto-complete__menu-select" tabindex="-1" auto-complete-cover="">
    <div class="var-input var--box">
      <div class="var-field-decorator var--box var-field-decorator--standard">
        <div class="var-field-decorator__controller var-field-decorator--error" style="cursor: text; overflow: hidden; --field-decorator-middle-offset-left: 0px; --field-decorator-middle-offset-width: 0px; --field-decorator-middle-offset-height: 0px;">
          <div class="var-field-decorator__icon"></div>
          <div class="var-field-decorator__middle">
            <!--v-if--><input id="var-input-mock-id" class="var-input__input" autocomplete="off" type="text">
          </div>
          <!--v-if-->
          <div class="var-field-decorator__icon">
            <!--v-if-->
          </div>
        </div>
        <div class="var-field-decorator__line var-field-decorator--line-error">
          <div class="var-field-decorator__dot var-field-decorator--line-error"></div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <transition-stub name="var-form-details" appear="false" persisted="false" css="true">
    <div class="var-form-details">
      <div class="var-form-details__error-message">
        <transition-stub name="var-form-details__message" appear="false" persisted="false" css="true">
          <div>Required</div>
        </transition-stub>
      </div>
      <div class="var-form-details__extra-message">
        <transition-stub name="var-form-details__message" appear="false" persisted="false" css="true">
          <!--v-if-->
        </transition-stub>
      </div>
    </div>
  </transition-stub>
</div>"
`;

exports[`auto-complete validation 2`] = `
"<div class="var-auto-complete" tabindex="0">
  <div class="var-menu var--box var-menu-select var-auto-complete__menu-select" tabindex="-1" auto-complete-cover="">
    <div class="var-input var--box">
      <div class="var-field-decorator var--box var-field-decorator--standard">
        <div class="var-field-decorator__controller" style="cursor: text; overflow: hidden; --field-decorator-middle-offset-left: 0px; --field-decorator-middle-offset-width: 0px; --field-decorator-middle-offset-height: 0px;">
          <div class="var-field-decorator__icon"></div>
          <div class="var-field-decorator__middle">
            <!--v-if--><input id="var-input-mock-id" class="var-input__input" autocomplete="off" type="text">
          </div>
          <!--v-if-->
          <div class="var-field-decorator__icon">
            <!--v-if-->
          </div>
        </div>
        <div class="var-field-decorator__line">
          <div class="var-field-decorator__dot"></div>
        </div>
      </div>
      <!--v-if-->
    </div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <transition-stub name="var-form-details" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
</div>"
`;

exports[`auto-complete variant 1`] = `
"<div class="var-auto-complete" tabindex="0">
  <div class="var-menu var--box var-menu-select var-auto-complete__menu-select" tabindex="-1" auto-complete-cover="">
    <div class="var-input var--box">
      <div class="var-field-decorator var--box var-field-decorator--outlined">
        <div class="var-field-decorator__controller" style="cursor: text; overflow: hidden; --field-decorator-middle-offset-left: 0px; --field-decorator-middle-offset-width: 0px; --field-decorator-middle-offset-height: 0px;">
          <div class="var-field-decorator__icon"></div>
          <div class="var-field-decorator__middle">
            <!--v-if--><input id="var-input-mock-id" class="var-input__input" autocomplete="off" type="text">
          </div>
          <!--v-if-->
          <div class="var-field-decorator__icon">
            <!--v-if-->
          </div>
        </div>
        <fieldset class="var-field-decorator__line">
          <legend class="var-field-decorator__line-legend">
            <!--v-if-->
          </legend>
        </fieldset>
      </div>
      <!--v-if-->
    </div>
    <!--teleport start-->
    <!--teleport end-->
  </div>
  <transition-stub name="var-form-details" appear="false" persisted="false" css="true">
    <!--v-if-->
  </transition-stub>
</div>"
`;
